src/util/triplea/MapXMLCreator/CanalDefinitionsPanel.java

```diff
@@ -0,0 +1,442 @@
+package util.triplea.MapXMLCreator;
+
+import java.awt.Color;
+import java.awt.FontMetrics;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import java.awt.event.ActionEvent;
+import java.awt.event.MouseEvent;
+import java.awt.geom.Rectangle2D;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import javax.swing.AbstractAction;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+
+import games.strategy.util.Tuple;
+import util.triplea.MapXMLCreator.TerritoryDefinitionDialog.DEFINITION;
+
+
+public class CanalDefinitionsPanel extends ImageScrollPanePanel {
+  private static final String NEW_CANAL_OPTION = "<new Canal>";
+
+  private SortedSet<String> selectedLandTerritories = new TreeSet<String>();
+  private SortedSet<String> selectedWaterTerritories = new TreeSet<String>();
+  private String currentCanalName = null;
+
+  private CanalDefinitionsPanel() {}
+
+  public static void layout(final MapXMLCreator mapXMLCreator, final JPanel stepActionPanel) {
+    ImageScrollPanePanel.mapXMLCreator = mapXMLCreator;
+    final CanalDefinitionsPanel panel = new CanalDefinitionsPanel();
+    panel.layout(stepActionPanel);
+    mapXMLCreator.setAutoFillAction(new AbstractAction() {
+      private static final long serialVersionUID = -8508734371454749752L;
+
+      @Override
+      public void actionPerformed(ActionEvent arg0) {
+        final int prevCanalCount = MapXMLHelper.canalDefinitions.size();
+        if (prevCanalCount > 0) {
+          if (JOptionPane.YES_OPTION != JOptionPane.showOptionDialog(null,

```

Comment:
There might be a helper method in the project file `SwingComponents`, if not there should be one created there to simplify this type of thing.